classes/Entities/HelpdeskProduct.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Avado\MoodleAbstractionLibrary\Entities;
+
+/**
+ * Class HelpdeskProduct
+ * @package Avado\MoodleAbstractionLibrary\Entities
+ */
+class HelpdeskProduct extends BaseModel
+{
+    /**
+     * @var string
+     */
+    protected $table = 'helpdesk_products';
+
+    /**
+     * Get the course_categories id, via the helpdesk_product_courses table
+     *
+     * @return \Illuminate\Database\Eloquent\Relations\HasManyThrough
+     */
+    public function productCategories(){
+        return $this->hasManyThrough(HelpdeskProduct::class, HelpdeskProductCourse::class, 'coursecategoryid', 'id', 'id', 'id');
```

Comment:
if I am not wrong then, you can use "hasMany" relationship as you want only "course_categories id". "helpdesk_product_courses" table can have multiple "productid", using hasmany you can get "coursecategoryid" associated with that product.